contents/tutorials/nextjs-ab-tests.md

```diff
@@ -1,269 +1,283 @@
+To track our app and set up an A/B test, we install PostHog. If you don't have a PostHog instance already, you can [sign up for free](https://us.posthog.com/signup). 
+
+Start by installing the `posthog-js` SDK:
 
 ```bash
-npm install posthog-js
+npm i posthog-js
 ```
 
-In `_app.js`, we can then initialize PostHog and set up a provider that provides access to PostHog throughout our app. 
+Next, create a `providers.js` file in your `app` folder. In it, initialize PostHog with your project API key and instance address and export a provider component.
 
 ```js
-// pages/_app.js
-import { PostHogProvider } from 'posthog-js/react'
+// app/providers.js
+'use client'
 import posthog from 'posthog-js'
+import { PostHogProvider } from 'posthog-js/react'
 
-if (typeof window !== 'undefined') {
-  posthog.init('<ph_project_api_key>', 
-  { 
-    api_host: '<ph_instance_address>'
-  })
-}
+export function PHProvider({ children }) {
+	if (typeof window !== 'undefined') {
+	  posthog.init('<ph_project_api_key>', {
+	    api_host: '<ph_instance_address>'
+	  })
+	}
 
-export default function App({ Component, pageProps }) {
-  return (
-    <PostHogProvider client={posthog}>
-      <Component {...pageProps} />
-    </PostHogProvider>
-  )
+  return <PostHogProvider client={posthog}>{children}</PostHogProvider>
 }
 ```
 
-Once you’ve done this, reload your app and click the button a few times. You should see this being captured in PostHog.
-
-![Events](../images/tutorials/nextjs-ab-tests/events.png)
-
-## Creating an action for our experiment goal
-
-The first part of setting up our experiment in PostHog is setting up the goal. In this case, it is clicks of the main call to action button which has varying content.
-
-To measure this, we can set up an [action](/manual/actions) using the PostHog toolbar. To enable and launch the toolbar, go to the "Launch Toolbar" tab in PostHog, add `http://localhost:3000/` as an authorized URL, then click launch. This brings you to your app with a PostHog icon hovering on your screen.
+Once created, you can import `PHProvider` into your `layout.js` file and wrap your app in it:
 
-To create an action with the toolbar, click:
+```js
+import "./globals.css";
+import { PHProvider } from './providers'
 
-1. the PostHog icon
-2. "Inspect" 
-3. the "Click Me" button, 
-4. "Create New Action" in the modal. 
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <PHProvider>
+        <body>{children}</body>
+      </PHProvider>
+    </html>
+  );
+}
+```
 
-Name the action "Clicked Main CTA" and unselect the text match (because it changes), then scroll to the bottom of the modal to create an action.
+When you reload your app, you should see events captured in [your activity tab](https://us.posthog.com/events) in PostHog.
 
-![Action](../images/tutorials/nextjs-ab-tests/action.gif)
+## 3. **Creating an action for our experiment goal**
```

Comment:
Nit: I would simplify by just calling posthog.capture direclty, as doing it this way introduces 3 new concepts: actions, autocapture, and toolbar. It's a bit of a mental overhead to wrap your head around them